### PR TITLE
Implement history charts for Profile Overlay

### PR DESCRIPTION
**Pull request by EVAST9919**
Examples in the comment below.

Used source about ticks distribution:
https://stackoverflow.com/questions/4947682/intelligently-calculating-chart-tick-positions

### CODE REVIEW

File: osu.Game/Overlays/Profile/Sections/Historical/ProfileLineChart.cs [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+            }
+        }
+
+        private void createColumnTicks()
+        {
+            columnTicksContainer.Clear();
+            columnLinesContainer.Clear();
+
+            var min = values.Select(v => v.Date).Min().ToOADate();
+            var max = values.Select(v => v.Date).Max().ToOADate();
```
Comment (bdach) [inline osu.Game/Overlays/Profile/Sections/Historical/ProfileLineChart.cs:157]: empty sequences are not checked here and will crash.

also calculating the difference would be done more sanely not by taking this eldritch `ToOADate` thing (why is that even a thing?) but via `TimeSpan.TotalDays` or similar.

**Review by bdach — DISMISSED** [review]

**Comment by EVAST9919** [comment]
After refactoring ticks are much closer to the web
Examples:
1.
web1
pr1
2.
web2
pr2
3.
web3
pr3
4.
web4
pr4

**Comment by bdach** [comment]
can this be re-reviewed yet? some issues seem to be unresolved still.

**Comment by EVAST9919** [comment]
Nope, not yet

File: osu.Game/Overlays/Profile/Sections/Historical/ChartProfileSubsection.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+        protected override Drawable CreateContent() => new Container
+        {
+            RelativeSizeAxes = Axes.X,
+            AutoSizeAxes = Axes.Y,
+            Padding = new MarginPadding
+            {
+                Top = 10,
+                Left = 20,
+                Right = 40
```
[the remark this answers is not in the document]
Comment (bdach) [reply osu.Game/Overlays/Profile/Sections/Historical/ChartProfileSubsection.cs:30]: i guess web doesn't have an exact correspondent and these look close enough

**Comment by bdach** [comment]
made several follow-up changes here myself. i believe this to be mostly ready to go.

**Review by peppy — APPROVED** [review]

**Pull request merged**